src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRSpark.java

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines;
+
+import com.google.cloud.dataflow.sdk.transforms.SerializableFunction;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.ReadContextData;
+import org.broadinstitute.hellbender.engine.spark.AddContextDataToReadSpark;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.engine.spark.JoinStrategy;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink;
+import org.broadinstitute.hellbender.engine.spark.datasources.VariantsSparkSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.ApplyBQSRArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.transforms.ApplyBQSRSparkFn;
+import org.broadinstitute.hellbender.tools.spark.transforms.BaseRecalibratorSparkFn;
+import org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadsWriteFormat;
+import org.broadinstitute.hellbender.utils.recalibration.BaseRecalibrationEngine;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationArgumentCollection;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationReport;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.IOException;
+import java.util.List;
+@CommandLineProgramProperties(
+        summary = "This tools performs 2 steps of BQSR - crection of recalibration tables and rewriting of the bam, without writing the tables to disk. ",
+        oneLineSummary = "Both steps of BQSR",
+        usageExample = "BQSRSpark -I in.bam --BQSRKnownVariants in.vcf -O out.bam",
+        programGroup = SparkProgramGroup.class
+)
+/**
+ * BQSR. The final result is analysis-ready reads.
+ */
+public final class BQSRSpark extends GATKSparkTool {
```

Comment:
It is very confusing to have both `BQSRSpark` and `BaseRecalibratorSpark`. Can you name this something like `BQSRSparkPipeline` (since it runs two tools)?